tests/sentry/tasks/integrations/github/test_language_parsers.py

```diff
@@ -182,7 +184,168 @@ def test_javascript_simple(self):
+
+@@44,38@@ function hasInnerFunction2(a: string) { sayHi();
+
+@@44,38@@ function hasInnerFunction3(a: string, b: string) { sayHi(); }
+
+@@44,38@@ var varFunction1 = function() { sayHi() };
+
+@@44,38@@ var varFunction2 = function() {
+
+@@44,38@@ var varFunction3 = function eval() { };
+
+@@44,38@@ var varFunction4 = function arguments(
+
+@@44,38@@ var varFunction5 = function hi() { sayHi() };
+
+@@44,38@@ function protoFunction(__proto__) { }
+
+@@44,38@@ function test() { "use strict" + 42; }
+
+@@44,38@@ function a(x: string, x: int) {'use strict';}
+
+@@44,38@@ function hello() { sayHi(); }
+
+@@44,38@@ function lol() { }
+"""
+
+        assert JavascriptParser.extract_functions_from_patch(patch) == {
+            "noArguments1",
+            "noArguments2",
+            "test1",
+            "test2",
+            "test3",
+            "hasInnerFunction1",
+            "hasInnerFunction2",
+            "hasInnerFunction3",
+            "varFunction1",
+            "varFunction2",
+            "varFunction3",
+            "varFunction4",
+            "varFunction5",
+            "protoFunction",
+            "test",
+            "a",
+            "hello",
+            "lol",
+        }
+
+    # reference https://github.com/jquery/esprima/tree/main/test/fixtures/ES6/arrow-function
+    # tests arrow functions on the same line as the hunk header
+    patch = """\
+@@44,38@@ export const arrow1 = (...a) => {
+
+@@44,38@@ var arrow2 = (a, ...b) => 0;
+
+@@44,38@@ var arrow3 = ((a)) => 0
+
+@@44,38@@ const arrow4 = (sun: planet) => earth
+
+@@44,38@@ const arrow5 = foo((x: a, y: b) => {})
+
+@@44,38@@ const arrow6 = foo(() => {})
+
+@@44,38@@ const arrow7 = (x: string) => ((y, z) => (x, y, z))
+
+@@44,38@@ const arrow8 = x => y => 42
+
+@@44,38@@ const arrow9 = (x: () => {}) => y => 42
+
+@@44,38@@ const arrow10 = (x => x)
+
+@@44,38@@ const arrow11 = (eval: string, a: int = 10) => 42
+
+@@44,38@@ const arrow12 = (eval:int = 10) => 42
+
+@@44,38@@ const arrow13 = (eval: int, a: () => void) => 42
+
+@@44,38@@ const arrow14 = (a: int) => 00
+
+@@44,38@@ const arrow15 = arguments => 42
+
+@@44,38@@ const arrow16 = (x: int=1) => x * x
+
+@@44,38@@ const arrow17 = (a: int, b: int) => { 42; }
+
+@@44,38@@ const arrow18 = e => { label: 42 }
+
+@@44,38@@ const arrow19 = e => ({ property: 42 })
+
+@@44,38@@ const arrow20 = e => { 42; }
+
+@@44,38@@ const arrow21 = (a: () => void, b: int) => "test"
+
+@@44,38@@ const arrow22 = (e: string) => "test"
+
+@@44,38@@ const arrow23 = e => e + 1
+
+@@44,38@@ const arrow24 = e => "test"
+
+@@44,38@@ const arrow25 = () => "test"
+
+@@44,38@@ var arrow26 = (...a) => 0
+
+@@44,38@@ var arrow27 = (...a, ...b) => 0
+
+@@44,38@@ var arrow28 = (a: string,b: int,...c) => 0;
+
+@@44,38@@ var arrow29 = (a ...b) => 0
+"""
+
+    functions = JavascriptParser.extract_functions_from_patch(patch)
+
+    for i in range(1, 30):
+        assert "arrow" + str(i) in functions
```

Comment:
Hmmm, this is mostly just testing the same thing over and over again no? What are we really trying to make sure here? That the different definitions of the arrow functions get parsed? If so, we should just check the length of the array and that should be enough.

But if we're trying to make sure all the different type of function names can be parsed, we should make different test cases, like `someFunctionLikeThis` and `ANOTHER_FUNCTION_LIKE_THIS` and `wH4t-IF-I-d000-this`, and assert that those names are found.

Reply:
it's all of the different ways you can create an arrow function, and that we get the name of the function correctly